### PR TITLE
emit('error' when loops crash

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

**Comment by jonasfj** [comment]
@imbstack don't you think this will do?

In production nobody is listening for `on('error,... )` so it should be cause monitor to report to sentry and crash...

If I ever wanted to I could listen for the error in tests... though that won't be relevant..

**Comment by imbstack** [comment]
r+ once the tests pass

File: src/dependencyresolver.js [hunk 1]
```diff
@@ -88,9 +93,10 @@ class DependencyResolver {
         try {
           await this.dependencyTracker.resolveTask(m.taskId, m.resolution);
           await m.remove();
+          this.monitor.count('handled-messages-success', 1);
         } catch (err) {
-          debug("[alert-operator] Failed to handle message: %j" +
-                ", with err: %s, as JSON: %j", message, err, err, err.stack);
+          this.monitor.count('handled-messages-error', 1);
+          this.reportError(err, 'warning');
```
Comment (gregarndt) [inline src/dependencyresolver.js]: should this be this.monitor.reportError?

Comment (gregarndt) [inline src/dependencyresolver.js]: Does reporting an error cause things to crash then?

Comment (jonasfj) [inline src/dependencyresolver.js]: monitor.reportError is much better :)

Comment (jonasfj) [inline src/dependencyresolver.js]: > Does reporting an error cause things to crash then or is the goal to just catch this error, report it, and not crash?

Nope, but `.emit('error', err)` does... This is just a warning... the loop continues...

**Pull request merged**